src/quickgui/qgsquickmaptransform.h

```diff
@@ -0,0 +1,75 @@
+/***************************************************************************
+  qgsquickmaptransform.h
+  --------------------------------------
+  Date                 : 27.12.2014
+  Copyright            : (C) 2014 by Matthias Kuhn
+  Email                : matthias (at) opengis.ch
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#ifndef QGSQUICKMAPTRANSFORM_H
+#define QGSQUICKMAPTRANSFORM_H
+
+#include <QQuickItem>
+#include <QMatrix4x4>
+
+#include "qgis_quick.h"
+
+class QgsQuickMapSettings;
+
+/**
+ * \ingroup quick
+ * The QgsQuickMapTransform is transformation that can be attached to any QQuickItem.
+ *
+ * If the item is based on the map coordinates, QgsQuickMapTransform will
+ * transform it to the device coordintes based on the attached map settings.
+ *
+ * \note QML Type: MapTransform
+ *
+ * \since QGIS 3.4
+ */
+class QUICK_EXPORT QgsQuickMapTransform : public QQuickTransform
+{
+    Q_OBJECT
+
+    //! map settings
+    Q_PROPERTY( QgsQuickMapSettings *mapSettings READ mapSettings WRITE setMapSettings NOTIFY mapSettingsChanged )
+
+  public:
+    //! create new map transform
+    QgsQuickMapTransform() = default;
+    ~QgsQuickMapTransform() = default;
+
+    /**
+     * Apply transformation based on current map settings to a matrix.
+     *
+     * Also optimize resulting matrix after transformation
+     * \param matrix Matrix to be transformed
+     */
+    void applyTo( QMatrix4x4 *matrix ) const;
+
+    //! \copydoc QgsQuickMapTransform::mapSettings
+    QgsQuickMapSettings *mapSettings() const;
+
+    //! \copydoc QgsQuickMapTransform::mapSettings
+    void setMapSettings( QgsQuickMapSettings *mapSettings );
+
+  signals:
+    //! \copydoc QgsQuickMapTransform::mapSettings
```

Comment:
FYI, sipify doesn't understand copydoc (unfortunately). It shouldn't matter here (for now) but it's good to keep this in mind.
At the same time, the documentation which is copied here is even shorter than the copydoc command and really only states the bare minimal obvious.